backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/objcexport/ObjCExportCodeGenerator.kt

```diff
@@ -600,15 +605,10 @@ private fun ObjCExportCodeGenerator.generateContinuationToCompletionConverter(
     }
 }
 
-private const val maxConvertorsInCache = 33
-
 private fun ObjCExportBlockCodeGenerator.emitFunctionConverters() {
     require(context.producedLlvmModuleContainsStdlib)
-    var count = context.ir.symbols.functionIrClassFactory.builtFunctionNClasses.size
-    // TODO: ugly hack to avoid huge unneeded adaptors linked into every binary, needs rework.
-    if (context.config.produce.isCache) {
-        count = count.coerceAtMost(maxConvertorsInCache)
-    }
+    val count = max(context.ir.symbols.functionIrClassFactory.builtFunctionNClasses.size,
```

Comment:
Not sure if this is correct.
This code takes "at most first 33 generated function types", while it should take "all function types with <= N parameters".